### ISSUE TITLE
heap.c "maybe uninitialized" compilation error under GCC 8

### ISSUE THREAD

**Issue by d** [greenplum-db/gpdb#6121]
### Greenplum version or build
master (6.0 alpha as of commit ac6d1f741620e9efe55af1a5b835f8088f64e545)

### OS version and uname -a
Ubuntu 18.10, although any platform where GCC 8 runs will reproduce this error

### autoconf options used ( config.status --config )
`./configure --enable-orca --with-gssapi --enable-mapreduce --with-perl --with-libxml --with-python --disable-gpcloud --disable-pxf --enable-gpfdist --enable-debug`

### Expected behavior
It compiles.

### Actual behavior
```
heap.c: In function 'heap_create_with_catalog':
heap.c:1675:3: error: 'stdRdOptions' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   InsertAppendOnlyEntry(relid,
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
          stdRdOptions->blocksize,
          ~~~~~~~~~~~~~~~~~~~~~~~~
          safefswritesize,
          ~~~~~~~~~~~~~~~~
          stdRdOptions->compresslevel,
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          stdRdOptions->checksum,
          ~~~~~~~~~~~~~~~~~~~~~~~
                               stdRdOptions->columnstore,
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
          stdRdOptions->compresstype,
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          InvalidOid,
          ~~~~~~~~~~~
          InvalidOid,
          ~~~~~~~~~~~
          InvalidOid,
          ~~~~~~~~~~~
          InvalidOid,
          ~~~~~~~~~~~
          InvalidOid);
          ~~~~~~~~~~~
cc1: some warnings being treated as errors
make[3]: *** [<builtin>: heap.o] Error 1
```

**Comment by d** [comment]
@hlinnaka I heard you finally joined the GCC 8 club! Did you not get this warning/error?

**Comment by hlinnaka** [comment]
Hmm, I seem to have "-Wno-error=maybe-uninitialized" in my CFLAGS. I don't remember why I added that, but that's why I haven't run into those warnings..

**Comment by hlinnaka** [comment]
After removing "-Wno-error=maybe-uninitialized", I'm still not seeing that. But I am seeing this:
```
resgroupcmds.c: In function ‘AlterResourceGroup’:
../../../src/include/c.h:852:4: error: ‘cpuset’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
    strncpy(_dst, (src), _len); \
    ^~~~~~~
resgroupcmds.c:374:14: note: ‘cpuset’ was declared here
  const char *cpuset;
              ^~~~~~
```

**Comment by d** [comment]
> After removing "-Wno-error=maybe-uninitialized", I'm still not seeing that. But I am seeing this:

Did you clear out your `CFLAGS`? Alternatively, try to use `-O3` in your `CFLAGS` for autoconf

**Comment by d** [comment]
For context, I don't see this at `-O0` or `-O1`

**Comment by hlinnaka** [comment]
Ah, with -O3, I'm seeing that same waring/error you did. And that resgroupcmds.c warning only happens with -O1.

**Comment by hlinnaka** [comment]
Fixed by commit b64ec2ec7f.
